src/utils/crypto.ts

```diff
@@ -0,0 +1,135 @@
+import type { IKey } from '@/types';
+import { STORAGE_KEYS, PASSWORD_ENCRYPTION_ALGO, IS_EXTENSION } from '@/constants';
+import { WalletStorage } from '@/lib/WalletStorage';
+import { toRaw } from 'vue';
+
+/**
+ * If the application is running as an extension, the key should be extractable.
+ * So that we can save it in the session storage.
+ */
+const IS_EXTRACTABLE = !!IS_EXTENSION;
+
+// Utility Functions for Base64 Encoding and Decoding
+const encodeBase64 = (data: Uint8Array) => Buffer.from(data).toString('base64');
+const decodeBase64 = (data: string) => Buffer.from(data, 'base64');
+
+export async function generateKey(password: string, encryptedMnemonic?: string): Promise<IKey> {
```

Comment:
I think the function that generates the key should not take the `encryptedMnemonic` as an argument. It makes the code hard to read and follow. Generating key should be a simple function. Consider providing the custom iv / salt as an `options` argument and have separate function that gets the salt and iv from encoded mnemonic.